tests/js/plugin/texthighlight-test.coffee

```diff
@@ -0,0 +1,78 @@
+assert = chai.assert
+sinon.assert.expose(assert, prefix: '')
+
+# In order to be able to create highlights,
+# the Annotator.TextHighlight class must exist.
+# This class is registered then the TextHighlights plugin
+# is initialized, so we will do that.
+th = new Annotator.Plugin.TextHighlights()
+th.pluginInit()
+
+describe 'Annotator.Plugin.TextHighlight', ->
+  sandbox = null
+  jqElement = null
+  scrollTarget = null
+
+  createTestHighlight = ->
+    anchor =
+      id: "test anchor"
+      annotation: "test annotation"
+      anchoring:
+        id: "test anchoring manager"
+        annotator:
+          id: "test annotator"
+          element:
+            delegate: sinon.spy()
+
+    new Annotator.TextHighlight anchor, "test page", "test range"
+
+  beforeEach ->
+    sandbox = sinon.sandbox.create()
+    sandbox.stub Annotator.TextHighlight, 'highlightRange',
+      (normedRange, cssClass) -> "test highlight span"
+
+    sandbox.stub Annotator.$.fn, "init", (selector, context) ->
```

Comment:
> As for mocking jquery, I don't think you have to. You've already
> mocked |highlightRange| so why not return an actual |Array| of
> |Element|, or jQuery collection, or whatever you need, rather than
> "test highlight span"? Then the jQuery calls would work on them. For
> |scrollIntoView| you could instead spy on or mock
> |$.fn.scrollintoview|, which is where the plugin is installed, and
> ensure that it gets called.

OK, I did that in
https://github.com/hypothesis/h/commit/d0c07d681c0fe7cb3f77b4f53bde53b94edf3153

How about now?